extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ChunkInputStreamGenerator.java

```diff
@@ -144,6 +149,36 @@ static <T> ChunkInputStreamGenerator makeInputStreamGenerator(
                 if (type == Short.class) {
                     return ShortChunkInputStreamGenerator.convertBoxed(chunk.asObjectChunk(), rowOffset);
                 }
+                if (type == LocalDate.class) {
+                    final long msPerDay = 24 * 60 * 60 * 1000L;
+                    final long minDate = QueryConstants.MIN_LONG / msPerDay;
+                    final long maxDate = QueryConstants.MAX_LONG / msPerDay;
+                    return LongChunkInputStreamGenerator.<LocalDate>convertWithTransform(chunk.asObjectChunk(),
+                            rowOffset, date -> {
+                                if (date == null) {
+                                    return QueryConstants.NULL_LONG;
+                                }
+                                final long epochDay = date.toEpochDay();
```

Comment:
https://arrow.apache.org/docs/python/generated/pyarrow.date64.html#pyarrow.date64 seems to suggest that this needs to be millis form the epoch, meaning we are doing the math too simply.

Reply:
This is the documentation source that matters which clearly indicates that leap seconds are ignored:
```
enum DateUnit: short {
  DAY,
  MILLISECOND
}

/// Date is either a 32-bit or 64-bit signed integer type representing an
/// elapsed time since UNIX epoch (1970-01-01), stored in either of two units:
///
/// * Milliseconds (64 bits) indicating UNIX time elapsed since the epoch (no
///   leap seconds), where the values are evenly divisible by 86400000
/// * Days (32 bits) since the UNIX epoch
table Date {
  unit: DateUnit = MILLISECOND;
}
```